packages/mattermost-redux/src/reducers/entities/threads/counts.ts

```diff
@@ -139,6 +139,15 @@ export function countsIncludingDirectReducer(state: ThreadsState['counts'] = {},
     case ChannelTypes.RECEIVED_CHANNEL_DELETED:
     case ChannelTypes.LEAVE_CHANNEL:
         return handleLeaveChannel(state, action, extra);
+    case ThreadTypes.RECEIVED_UNREAD_THREADS:
+        return {
```

Comment:
Not sure how often this might get called but I wonder if it's worth doing a compare here to see if there was any change, and if not then not updating state at all. Could potentially save a lot of selectors recalculating.

Reply:
@jwilander yes it would be updated each time we get threads,
I've added a shallow equality check on counts for both fetching unread and all threads,
thanks for pointing it out!